ImageViewJpw gives DeprecationWarning: There is no current event loop

Saw this downstream in astrowidgets pre-release testing. I am going to ignore this warning downstream but thought maybe it is something Ginga should handle?

xref https://github.com/astropy/astrowidgets/pull/156

Example log: https://github.com/astropy/astrowidgets/actions/runs/3658672508/jobs/6183750830

```
>       image = ImageWidget()

../../astrowidgets/tests/test_api.py:17: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
../../astrowidgets/core.py:104: in __init__
    self._viewer.set_widget(self._jup_img)
.../ginga/web/jupyterw/ImageViewJpw.py: in set_widget
    super(ImageViewEvent, self).set_widget(jp_imgw)
.../ginga/web/jupyterw/ImageViewJpw.py:96: in set_widget
    self._defer_task = JpHelp.Timer()
.../ginga/web/jupyterw/JpHelp.py:115: in __init__
    self._timer = self.timer_factory.add_timer(self._redirect_cb)
.../ginga/web/jupyterw/JpHelp.py:75: in add_timer
    self.wind()
.../ginga/web/jupyterw/JpHelp.py:43: in wind
    self._timeout = IOLoop.current().add_timeout(delta, self.timer_tick)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

instance = True

    @staticmethod
    def current(instance: bool = True) -> Optional["IOLoop"]:  # noqa: F811
        """Returns the current thread's `IOLoop`.
    
        If an `IOLoop` is currently running or has been marked as
        current by `make_current`, returns that instance.  If there is
        no current `IOLoop` and ``instance`` is true, creates one.
    
        .. versionchanged:: 4.1
           Added ``instance`` argument to control the fallback to
           `IOLoop.instance()`.
        .. versionchanged:: 5.0
           On Python 3, control of the current `IOLoop` is delegated
           to `asyncio`, with this and other methods as pass-through accessors.
           The ``instance`` argument now controls whether an `IOLoop`
           is created automatically when there is none, instead of
           whether we fall back to `IOLoop.instance()` (which is now
           an alias for this method). ``instance=False`` is deprecated,
           since even if we do not create an `IOLoop`, this method
           may initialize the asyncio loop.
    
        .. deprecated:: 6.2
           It is deprecated to call ``IOLoop.current()`` when no `asyncio`
           event loop is running.
        """
        try:
>           loop = asyncio.get_event_loop()
E           DeprecationWarning: There is no current event loop

.../tornado/ioloop.py:265: DeprecationWarning
```

--- a/doc/WhatsNew.rst
+++ b/doc/WhatsNew.rst
@@ -2,6 +2,10 @@
 What's New
 ++++++++++
 
+Ver 4.1.0 (unreleased)
+======================
+- fixed a DeprecationWarning with jupyterw back end
+
 Ver 4.0.0 (2022-12-20)
 ======================
 - fixed getattr functionality in Bunch

--- a/ginga/util/evloop.py
+++ b/ginga/util/evloop.py
@@ -1,0 +1,35 @@
+#
+# evloop.py -- asyncio/tornado event loop utilities.
+#
+# This is open-source software licensed under a BSD license.
+# Please see the file LICENSE.txt for details.
+#
+import sys
+import asyncio
+
+
+def get_event_loop():
+    """Get the current asyncio event loop."""
+    # workaround for DeprecationError in Python 3.10
+    if sys.version_info < (3, 10):
+        loop = asyncio.get_event_loop()
+    else:
+        try:
+            loop = asyncio.get_running_loop()
+        except RuntimeError:
+            loop = asyncio.new_event_loop()
+            asyncio.set_event_loop(loop)
+
+    return loop
+
+
+def get_ioloop():
+    """Get the current tornado ioloop."""
+    from tornado.ioloop import IOLoop
+
+    loop = get_event_loop()
+
+    # now we should be able to get the current tornado IOLoop,
+    # which as of Tornado > v5 is a wrapper around the asyncio
+    # event loop
+    return IOLoop.current()

--- a/ginga/web/jupyterw/JpHelp.py
+++ b/ginga/web/jupyterw/JpHelp.py
@@ -9,9 +9,8 @@ import random
 import datetime
 import threading
 
-from tornado.ioloop import IOLoop
-
 from ginga.misc import Bunch, Callback, log
+from ginga.util.evloop import get_ioloop
 
 default_timer_interval_msec = 10
 
@@ -40,7 +39,7 @@ class TimerFactory(object):
         # expiring at the same time
         interval = random.randint(1, self.base_interval_msec)  # nosec
         delta = datetime.timedelta(milliseconds=interval)
-        self._timeout = IOLoop.current().add_timeout(delta, self.timer_tick)
+        self._timeout = get_ioloop().add_timeout(delta, self.timer_tick)
 
     def timer_tick(self):
         """Callback executed every self.base_interval_msec to check timer
@@ -50,7 +49,7 @@ class TimerFactory(object):
         self.process_timers()
 
         delta = datetime.timedelta(milliseconds=self.base_interval_msec)
-        self._timeout = IOLoop.current().add_timeout(delta, self.timer_tick)
+        self._timeout = get_ioloop().add_timeout(delta, self.timer_tick)
 
     def process_timers(self):
         self.logger.debug("check timers")

--- a/ginga/web/pgw/PgHelp.py
+++ b/ginga/web/pgw/PgHelp.py
@@ -16,10 +16,10 @@ from io import BytesIO
 import tornado.web
 import tornado.websocket
 import tornado.template
-from tornado.ioloop import IOLoop
 
 from ginga.misc import Bunch, Callback
 from ginga.util.io import io_rgb
+from ginga.util.evloop import get_ioloop
 from ginga.fonts import font_asst
 
 font_regex = re.compile(r'^(.+)\s+(\d+)$')
@@ -92,7 +92,7 @@ class ApplicationHandler(tornado.websocket.WebSocketHandler):
         # expiring at the same time
         interval = random.randint(1, self.interval)  # nosec
         delta = datetime.timedelta(milliseconds=interval)
-        self.timeout = IOLoop.current().add_timeout(delta, self.timer_tick)
+        self.timeout = get_ioloop().add_timeout(delta, self.timer_tick)
 
     def add_event_type(self, msg_type, event_class):
         self.event_callbacks[msg_type] = event_class
@@ -101,7 +101,7 @@ class ApplicationHandler(tornado.websocket.WebSocketHandler):
         self.set_nodelay(True)
 
     def on_close(self):
-        IOLoop.current().remove_timeout(self.timeout)
+        get_ioloop().remove_timeout(self.timeout)
 
     def on_message(self, raw_message):
         message = json.loads(raw_message)
@@ -128,7 +128,7 @@ class ApplicationHandler(tornado.websocket.WebSocketHandler):
         self.app.widget_event(event)
 
         delta = datetime.timedelta(milliseconds=self.interval)
-        self.timeout = IOLoop.current().add_timeout(delta, self.timer_tick)
+        self.timeout = get_ioloop().add_timeout(delta, self.timer_tick)
 
 
 class WindowHandler(tornado.web.RequestHandler):

